docs/docs/config-reference.md

```diff
@@ -0,0 +1,242 @@
+
+### Identity Provider Client Secret
+
+- Environmental Variable: `IDP_CLIENT_SECRET`
+- Type: `string`
+- Required
+
+Client ID is the OAuth 2.0 Secret Identifier retrieved from your identity provider. For a detailed explanation of an OpenID Client Secret, see the [OIDC RFC Section 9](https://openid.net/specs/openid-connect-core-1_0.html#ClientAuthentication) or your identity provider's documentation. See [identity provider] for details.
+
+### Identity Provider URL
+
+- Environmental Variable: `IDP_PROVIDER_URL`
+- Type: `string`
+- Optional if your identity provider is a built-in IDP and has a static URL.
+
+Provider URL is the base path to an identity provider's [OpenID connect discovery document](https://openid.net/specs/openid-connect-discovery-1_0.html) For example, google's URL would be `https://accounts.google.com` for [their discover URL](https://accounts.google.com/.well-known/openid-configuration).
+
+### Identity Provider Scopes
+
+- Environmental Variable: `IDP_SCOPES`
+- Type: `[]string` comma separated list of oauth scopes.
+- Default: `oidc`,`profile`, `email`, `offline_access` (typically)
+- Optional for built-in identity providers.
+
+Identity provider scopes correspond to access privilege scopes as defined in Section 3.3 of OAuth 2.0 RFC6749\. The scopes associated with Access Tokens determine what resources will be available when they are used to access OAuth 2.0 protected endpoints. If you are using a built-in provider, you probably don't want to set customized scopes.
+
+## Proxy Service
+
+### Routes
+
+- Environmental Variable: `ROUTES`
+- Type: `map[string]string` comma separated mapping of gatewayed (reverse-proxied) entities.
+- Required
+- Example: `https://httpbin.corp.example.com=http://httpbin,https://hello.corp.example.com=http://hello:8080/`
+
+Routes contains a mapping of routes to be managed by pomerium.
+
+### Signing Key
+
+- Environmental Variable: `SIGNING_KEY`
+- Type: [base64 encoded] `string`
+- Optional
+
+Signing key is the base64 encoded key used to sign outbound requests. For more information see the [signed headers](./signed-headers.md) docs.
+
+### Authenticate Service URL
+
+- Environmental Variable: `AUTHENTICATE_SERVICE_URL`
+- Type: `URL`
+- Required
+- Example: `https://auth.corp.example.com`
+
+Authenticate Service URL is the externally accessible URL for the authenticate service.
+
+### Authenticate Internal Service URL
+
+- Environmental Variable: `AUTHENTICATE_INTERNAL_URL`
+- Type: `string`
+- Optional
+- Example: `pomerium-authenticate-service.pomerium.svc.cluster.local`
+
+Authenticate Service URL is the internal hostname for the authenticate service. Used in place of the authenticate service url for "behind-the-ingress communication. Setting is required for ingresses and load balancers that do not support HTTPS/2 or gRPC termination.
+
+### Authenticate Internal Service Address
```

Comment:
This doubles the documentation above in Authenticate Internal Service URL.

Reply:
Addressing in #53